src/Microsoft.DotNet.ProjectModel/ProjectReader.cs

```diff
@@ -246,6 +243,38 @@ public Project ReadProject(Stream stream, string projectName, string projectPath
             return project;
         }
 
+        internal static string NormalizeProjectDirectoryPath(string path)
+        {
+            if (File.Exists(path) &&
+                string.Equals(Path.GetFileName(path), Project.FileName, StringComparison.OrdinalIgnoreCase))
+            {
+                string directoryName = Path.GetDirectoryName(path);
+                if (string.IsNullOrEmpty(directoryName))
+                {
+                    directoryName = Directory.GetCurrentDirectory();
+                }
+
+                return Path.GetFullPath(directoryName);
+            }
+            else if (Directory.Exists(path) &&
+                     File.Exists(Path.Combine(path, Project.FileName)))
+            {
+                return Path.GetFullPath(path);
+            }
+
+            return null;
```

Comment:
The fact that "return string.Empty" occurs exactly 3 times in our current code base tells me that this isn't a general convention.

1. and one of those times is in a method that isn't even called.
2. And the other 2 are in the same method

Another reason is performane-wise returning `null` is a faster check for the caller.  You don't have to dereference the string and check for Length == 0.

> because it seems to me that return string.Empty; is a very small chance that should cost literally nothing do make.

I've already shown you that there are callers of this method that explicitly check for null.  So `cost literally nothing` doesn't apply here.